ui/js/directives/treeherder/clonejobs.js

```diff
@@ -166,87 +177,176 @@ treeherder.directive('thCloneJobs', [
             }, 200);
         };
 
+        /**
+         * Clicking a group will expand or collapse it.  Expanded shows all
+         * jobs.  Collapsed shows counts and failed jobs.
+         */
+        var clickGroupCb = function(el) {
+            var groupMap =  ThResultSetStore.getGroupMap($rootScope.repoName);
+            var gi = getGroupInfo(el, groupMap);
+            if (gi) {
+                if (isGroupExpanded(gi.jgObj)) {
+                    gi.jgObj.groupState = "collapsed";
+                    addGroupJobsAndCounts(gi.jgObj, gi.platformGroupEl);
+                } else {
+                    gi.grpCountList.empty();
+                    gi.jgObj.groupState = "expanded";
+                    addJobBtnEls(gi.jgObj, gi.grpJobList);
+                }
+            }
+        };
+
         var togglePinJobCb = function(ev, el, job){
             $rootScope.$emit(thEvents.jobPin, job);
         };
 
-        var addJobBtnEls = function(
-            jgObj, jobBtnInterpolator, jobTdEl){
-
-            var jobsShown = 0;
-
-            var lastJobSelected = ThResultSetStore.getSelectedJob(
-                $rootScope.repoName
-            );
-
-            var hText, key, resultState, job, jobStatus, jobBtn, l;
+        var addJobBtnEls = function(jgObj, jobList) {
+            var lastJobSelected = ThResultSetStore.getSelectedJob($rootScope.repoName);
+            var job, l;
             var jobBtnArray = [];
+            jobList.empty();
 
             for(l=0; l<jgObj.jobs.length; l++){
 
                 job = jgObj.jobs[l];
 
-                //Set the resultState
-                resultState = thResultStatus(job);
 
                 job.searchStr = thJobSearchStr(job) + ' ' + job.ref_data_name  + ' ' +
                     job.signature;
 
-                //Make sure that filtering doesn't effect the resultset counts
-                //displayed
-                if(thJobFilters.showJob(job) === false){
-                    //Keep track of visibility with this property. This
-                    //way down stream job consumers don't need to repeatedly
-                    //call showJob
-                    job.visible = false;
-                } else {
-                    jobsShown++;
-                    job.visible = true;
-                }
+                //Keep track of visibility with this property. This
+                //way down stream job consumers don't need to repeatedly
+                //call showJob
```

Comment:
This is incredibly nit-y, but perhaps we could take the opportunity to space legacy comments nicely. eg.

`// This with a leading space`
`// and this with a leading space`

I guess it would be nice to apply them to any comments involved in the change.